File: server/src/main/scala/kpn/api/common/ChangeSetSummaryNetworkInfo.scala
package kpn.api.common

case class ChangeSetSummaryNetworkInfo(
  networkChanges: NetworkChanges,
  routeChanges: Seq[ChangeSetSubsetElementRefs], // orphan route changes
  nodeChanges: Seq[ChangeSetSubsetElementRefs], // orphan node changes
)

File: backend/api/src/main/java/kpn/api/common/ChangeSetSummaryNetworkInfo.java
package kpn.api.common;

import kpn.api.common.ChangeSetSubsetElementRefs;
import kpn.api.common.NetworkChanges;

import com.google.common.collect.ImmutableList;

public record ChangeSetSummaryNetworkInfo(
  NetworkChanges networkChanges,
  ImmutableList<ChangeSetSubsetElementRefs> orphanRouteChanges,
  ImmutableList<ChangeSetSubsetElementRefs> orphanNodeChanges
) {
}
